src/components/Icon/tests/Icon.test.tsx

```diff
@@ -11,4 +16,32 @@ describe('<Icon />', () => {
       expect(element.prop('aria-label')).toBe('This is an icon');
     });
   });
+  describe('source', () => {
+    it("renders a placeholder div when source is set to 'placeholder'", () => {
+      const element = mountWithAppProvider(<Icon source="placeholder" />);
+      expect(findByTestID(element, 'placeholder').exists()).toBe(true);
+    });
+    it('renders an SVG when source is given a BundledIcon', () => {
+      const element = shallowWithAppProvider(<Icon source="add" />);
+      expect(element.find('svg').exists()).toBe(true);
```

Comment:
We use `length` over `exists` -> [guidelines](https://github.com/Shopify/web-foundation/blob/master/Best%20practices/Enzyme.md#Expectations)
```suggestion
      expect(element.find('svg')).toHaveLength(1);
```